File: Seojin/마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    int ans = Integer.MAX_VALUE;
    void btr(String storey, int sum, int cnt,int flag){
        if (ans<sum) return;
        if (cnt == 0){
            if (flag == 1){
                sum++;
            }
            ans = Math.min(ans,sum);
            return;
        }
        int cur = storey.charAt(cnt-1)-'0';
        if (flag == 1){
            if (cur == 9){
                cur = 0;
                btr(storey,sum+cur, cnt-1,1);
                btr(storey,sum+(10-cur), cnt-1,1); 
            }
            else {
                cur++;
                btr(storey,sum+cur, cnt-1,0);
                btr(storey,sum+(10-cur), cnt-1,1);
            }
        }
        else {
            btr(storey,sum+cur, cnt-1,0);
            btr(storey,sum+(10-cur), cnt-1,1); 
        }

        
    }
    public int solution(int storey) {
        int answer = 0;
        String s_storey = String.valueOf(storey);

        btr(s_storey,0,s_storey.length(),0);
        
        answer = ans;
        return answer;
    }
}

File: Suho/마법의 엘리베이터/마법의 엘리베이터.py
def solution(storey):
    answer = 0
    #그리디 같음
    #5보다 크면, + / 5보다 작으면 -
    while True:
        R=storey%10
        storey=storey//10
        if R<5:
            #빼기
            answer +=R
        elif R > 5:
            #더하기
            answer += (10-R)
            storey= storey+1
        else: #R==5인 경우
            Q=storey %10
            if Q<5:
                answer +=R
            else:
                answer += (10-R)
                storey= storey+1
        
        if storey == 0:
            break
    return answer